### DISCUSSION TITLE
Is there any option to disable clip path ?

### DISCUSSION THREAD

**Discussion by CenkErdonmez** [katspaugh/wavesurfer.js#3558] [Q&A]
Hey , 

We are trying to create a app using wavesurfer.js , everything going great but we also want people to select regions from a video they upload . We managed to create frames set regions bla bla .. The problem we are having at the moment is that whenever we play the video because of the renderProgress function in node modules background of the canvas changes so we lose the frames.  What we need is to disable clip path option that comes with the renderProgress function. 

Hope someone can help.

**Comment by katspaugh** [comment]
There's no such option but we could refactor the code to not use clip-path. It was added here: https://github.com/katspaugh/wavesurfer.js/pull/3285

**Reply by CenkErdonmez** [reply]
Thanks for the replay. We find a work around. Not the best way to fix it but we manage to manipulate the dom(you can see the code we used at the end) (sometimes it looks a little laggy though) . If there is any chance to add this as an option so we can pass it through when we need to draw frames to canvas it would be superb if no than no need to waste your time on it 😄 


`const canvases = document.querySelector(`#waveform-${mediaId} > div`).shadowRoot.querySelector('div > div > div.canvases');`
`canvases.style.clipPath = 'none';`

**Reply by katspaugh** [reply]
Could you post a screenshot of how it looks with the clip-path still in place? And also a screenshot of the Dom tree with your video inserted. Just trying to understand how the progress div overlaps with it.

**Reply by CenkErdonmez** [reply]
Sorry for the late response gonna share few screenshots about it, ignore my drawing skills 😄 . 
![Ekran görüntüsü 2024-02-19 092905](https://github.com/katspaugh/wavesurfer.js/assets/113177005/30370fcc-e02c-4b33-a96d-7ced03a3836d)
![Ekran görüntüsü 2024-02-19 093141](https://github.com/katspaugh/wavesurfer.js/assets/113177005/3d0aeb9f-74c1-4ca4-84eb-44085806953b)
![Ekran görüntüsü 2024-02-19 093314](https://github.com/katspaugh/wavesurfer.js/assets/113177005/819332e5-c82b-4e9f-a7b9-a763043aec29)

On screenshots you can see with clip path on and off. Also posted the elements without clip path.

**Reply by katspaugh** [reply]
Gotcha, thanks! I'll see if we can make it optional, or remove altogether.

By the way, does it still clip if `progressColor: 'transparent'` is set?

**Reply by CenkErdonmez** [reply]
Just tried with it still clip path removes frames.
